Kernel/FileSystem/CustodyBase.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/RefPtr.h>
+#include <AK/StringView.h>
+#include <Kernel/FileSystem/Custody.h>
+
+namespace Kernel {
+
+class CustodyBase {
```

Comment:
:bikeshed:, but isn't this a `CustodyResolver`?

Reply:
Not really?, because all the mentioned methods had a argument called `base`, so I figured this is just `CustodyBase`... which is passed and is "resolved" on demand, and not a resolver.

Reply:
Well, in that case the `Custody base` actually already was a base, since it was already resolved. In this case we also tack on a path (not the base, but the target file) and some advanced logic, which makes this more of a... unresolved base Custody?